src/common/utils/get_resolved_document_by_id.py

```diff
@@ -90,14 +76,12 @@ def resolve_references_in_entity(
     current_id: str | None,
     depth: int = 1,
     depth_count: int = 0,
```

Comment:
I don't see the point of all these default values. Removing them is perhaps not in the scope of this PR though